service/common/src/main/java/org/apache/polaris/service/admin/PolarisAdminService.java

```diff
@@ -200,6 +205,43 @@ private Optional<CatalogRoleEntity> findCatalogRoleByName(String catalogName, St
         .map(path -> CatalogRoleEntity.of(path.getRawLeafEntity()));
   }
 
+  private <T> Stream<T> loadEntities(
+      @Nonnull PolarisEntityType entityType,
+      @Nonnull PolarisEntitySubType entitySubType,
+      @Nullable PolarisEntity catalogEntity,
+      @Nonnull Function<PolarisBaseEntity, T> transformer) {
+    List<PolarisEntityCore> catalogPath;
+    long catalogId;
+    if (catalogEntity == null) {
+      catalogPath = null;
+      catalogId = 0;
+    } else {
+      catalogPath = PolarisEntity.toCoreList(List.of(catalogEntity));
```

Comment:
I think this could be simplified, `PolarisEntity.toCoreList()` returns `null` when the input is empty.

Reply:
Ah, yeah, makes sense, forgot about that NPE.